### ISSUE TITLE
Icons that use a UTF-8 glyph are incorrectly presented by assistive technologies

### ISSUE THREAD

**Issue by ppvg** [minvws/nl-rdo-manon#433]
From independent accessibility report, SC 2.4.6:

<blockquote lang="nl">
Op pagina <a href="https://minvws.github.io/nl-rdo-manon/documentation/components/accordion.html">components/accordion.html</a> staan uitklapbare elementen. Deze elementen zijn opgemaakt als knoppen. De toegankelijkheidsnaam van deze knoppen is bijvoorbeeld "Voorbeeld-onderwerp 2 v". Het pijltje omhoog of de letter "v" als pijl naar beneden wordt dus ook onderdeel van de toegankelijkheidsnaam. Dit is hier eerder een icoon. Aangezien er met het <code>aria-expanded</code>-attribuut al aangegeven wordt dat dit element in- en uitgeklapt kan worden, kan het pijltje of de letter "v" beter verborgen worden. Dit kan door er een span-element omheen te plaatsen en hierop het attribuut <code>aria-hidden="true"</code> op te plaatsen.

Dit komt bijvoorbeeld ook voor op pagina <a href="https://minvws.github.io/nl-rdo-manon/documentation/components/filter.html">components/filter.html</a>.
</blockquote>

~Presentational icons, such as the arrows on expand/collapse buttons of accordions and filters, should be `aria-hidden`. This means they should be moved into the markup, since it is not possible to apply `aria-hidden` to CSS pseudo-elements (`::before` and `::after`).~

Icons that use a pseudo-element (`::before` or `::after`) with a UTF-8 glyph as `content` may be incorrectly presented by assistive technologies (e.g. screen readers or braille displays). Presentational icons should be hidden, and informative icons should have a text alternative. The UTF-8 glyph should not be part of the element's accessible name.

This could be a achieved by adding `aria-hidden` or `aria-label` to the icon. However, this is not possible for icons that are added directly to another element as a pseudo-element.

One way to solve this is to move the icons into markup, to a separate element to which aria attributes can be applied. This could be a separate `<span>` with a pseudo-element, a figure element such as an inline SVG, or anything else.

### Edit (2025-08-06)

This ticket originally incorrectly implied that moving the icon into markup is the only solution. In reality the glyph itself is the problem, not the pseudo-element. I've updated the title and description to focus on the problem rather than propose a solution.

Another recently proposed way to solve this issue is to use pseudo-elements with a `mask-image` and an empty `content:""` instead of a UTF-8 glyph. That way, the pseudo-element does not add anything to the parent-element's accessible name.

**Comment by underdarknl** [comment]
Should we also look at implementing this using details and summary tags instead of buttons and divs? Or at least also apply the css for 'our' accordions onto those tags.

**Comment by ppvg** [comment]
We've had a report of these glyphs causing "an annoying Braille character", so this issue is confirmed to affect multiple assistive technologies.

**Comment by HeleenSG** [comment]
> Should we also look at implementing this using details and summary tags instead of buttons and divs? Or at least also apply the css for 'our' accordions onto those tags.

Out of scope for this ticket. But I've created a separate one as it's a good suggestion.
https://github.com/minvws/nl-rdo-manon/issues/1101
